# Palmer Penguins

***

The notebook will use the Palmer Pengiuns data set.

This data set can be found here on [GitHub](https://allisonhorst.github.io/palmerpenguins/)

This notebook contains 
1. Overview of the data set
2. Types of variables contained in the data set
3. Barchart & Histogram of a variable in the data set
4. Corelation between 2 variables in the data set

## Imports
***
We use pandas for the DataFrame data structure.

It allows us to view, analyse and manupilate data as required for our notebook.

In [2]:
# Data frames.
import pandas as pd

## Load data
***
Once we have imported pandas, we should load the data set into the dataframe


In [3]:
 df = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv")

## Inspect data and provide the overview of it
***


In [21]:
# Look at the first row.
df.iloc[0]


species                 Adelie
island               Torgersen
bill_length_mm            39.1
bill_depth_mm             18.7
flipper_length_mm        181.0
body_mass_g             3750.0
sex                       MALE
Name: 0, dtype: object

In [16]:
# See the first 10 rows.
first_10_rows = df.head(10)

first_10_rows


species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       3625.0  FEMALE  
7       4675.0    MALE  
8       3475.0     NaN  
9       4250.0     NaN

In [14]:
# See the bottom 10 rows

last_10_rows = df.tail(10)

last_10_rows

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
334  Gentoo  Biscoe            46.2           14.1              217.0   
335  Gentoo  Biscoe            55.1           16.0              230.0   
336  Gentoo  Biscoe            44.5           15.7              217.0   
337  Gentoo  Biscoe            48.8           16.2              222.0   
338  Gentoo  Biscoe            47.2           13.7              214.0   
339  Gentoo  Biscoe             NaN            NaN                NaN   
340  Gentoo  Biscoe            46.8           14.3              215.0   
341  Gentoo  Biscoe            50.4           15.7              222.0   
342  Gentoo  Biscoe            45.2           14.8              212.0   
343  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
334       4375.0  FEMALE  
335       5850.0    MALE  
336       4875.0     NaN  
337       6000.0    MALE  
338       4925.0  FEMALE  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE

In [20]:
# Unique values in first 2 colummns to understand number of species and islands.

unique_species = df['species'].unique()
unique_islands = df['island'].unique()

print(unique_species)
print(unique_islands)

['Adelie' 'Chinstrap' 'Gentoo']
['Torgersen' 'Biscoe' 'Dream']


In [17]:
# Type of each variable.

df.dtypes

species               object
island                object
bill_length_mm       float64
bill_depth_mm        float64
flipper_length_mm    float64
body_mass_g          float64
sex                   object
dtype: object

In [18]:
# Number of Rows and Columns

num_rows = len(df)
num_cols = len(df.columns)

print(f"Number of Rows: {num_rows}")
print(f"Number of Columns: {num_cols}")


Number of Rows: 344
Number of Columns: 7


In [6]:
# Describe the data set.
df.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

The data set contains data about 3 pengiun species across 3 islands
It contains 7 columns and 344 rows

Description of the data can be found below

|Heading           |Type             |Description                          |
|------------------|:----------------|:------------------------------------|
|species           |           object| Species of the pengiun              |
|island            |           object| Island where penguin is found       |
|bill_length_mm    |          float64| Length of the bill of the penguin   |
|bill_depth_mm     |          float64| Depth of the bill of the penguin    |
|flipper_length_mm |          float64| Length of the flipper of the penguin|
|body_mass_g       |          float64| Body mass of the penguin            |
|sex               |           object| Sex of the penguin                  |


Object indicates presense of string variable, while float64 indicates numerical presence

It must be noted that for some unique records there is data missing for singular and/or multiple data variables. 

The summary description of the numerical variables can be found below
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>bill_length_mm</th>
      <th>bill_depth_mm</th>
      <th>flipper_length_mm</th>
      <th>body_mass_g</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>342.000000</td>
      <td>342.000000</td>
      <td>342.000000</td>
      <td>342.000000</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>43.921930</td>
      <td>17.151170</td>
      <td>200.915205</td>
      <td>4201.754386</td>
    </tr>
    <tr>
      <th>std</th>
      <td>5.459584</td>
      <td>1.974793</td>
      <td>14.061714</td>
      <td>801.954536</td>
    </tr>
    <tr>
      <th>min</th>
      <td>32.100000</td>
      <td>13.100000</td>
      <td>172.000000</td>
      <td>2700.000000</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>39.225000</td>
      <td>15.600000</td>
      <td>190.000000</td>
      <td>3550.000000</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>44.450000</td>
      <td>17.300000</td>
      <td>197.000000</td>
      <td>4050.000000</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>48.500000</td>
      <td>18.700000</td>
      <td>213.000000</td>
      <td>4750.000000</td>
    </tr>
    <tr>
      <th>max</th>
      <td>59.600000</td>
      <td>21.500000</td>
      <td>231.000000</td>
      <td>6300.000000</td>
    </tr>
  </tbody>
</table>
</div>





***

### End